CI Visibility - dotnet agentless beta documentation (#13609)
DataDog/documentation: content/en/continuous_integration/setup_tests/dotnet.md

[Install the Datadog Agent to collect tests data][1].

<div class="alert alert-warning">

> Agentless mode is in beta. To test this feature, follow the <a href="/#agentless-beta">instructions</a> on this page.

Reviewer: ```suggestion
Agentless mode is in beta. To test this feature, follow the <a href="/continuous_integration/setup_tests/dotnet#agentless-beta">instructions</a> on this page.
```

[Install the Datadog Agent to collect tests data][1].

> <strong>Note:</strong> Agentless mode is currently on beta, if you want to test this feature follow the [instructions][5].

Reviewer 2: ```suggestion
<strong>Note:</strong> Agentless mode is currently in beta, if you want to test this feature follow the [instructions][5].
```

> <strong>Note:</strong> Agentless mode is currently in beta, if you want to test this feature follow the [instructions][5].

Reviewer: ```suggestion
<div class="alert alert-warning">
Agentless mode is in beta. To test this feature, follow the <a href="/#agentless-beta">instructions</a> on this page.
</div>
```

For more information about how to add spans and tags for custom instrumentation, see the [.NET Custom Instrumentation documentation][4].

## Agentless (Beta)

To instrument your test suite without requiring an agent, prefix your test command with `dd-trace ci run`, providing the [Datadog API key][6] as the `--api-key`, the name of the service or library under test as the `--dd-service` parameter, and the environment where tests are being run (for example, `local` when running tests on a developer workstation, or `ci` when running them on a CI provider) as the `--dd-env` parameter. For example:

{{< code-block lang="bash" >}}

> dd-trace ci run --api-key <API KEY> --dd-service=my-dotnet-app --dd-env=ci -- dotnet test

Reviewer 2: The wording in this section seems to recommend using the `--api-key` param, but I think we should be recommending the env var to avoid persisting the api key in the terminal's history.

Author: Changed in [f24b1da](https://github.com/DataDog/documentation/pull/13609/commits/f24b1daab93b8669fa6e3a73f36feffbef732c11)

For more information about how to add spans and tags for custom instrumentation, see the [.NET Custom Instrumentation documentation][4].

## Agentless (Beta)

> To instrument your test suite without requiring an agent, you need to configure the following environment variables:

Reviewer: ```suggestion
To instrument your test suite without requiring an Agent, configure the following environment variables:
```

`DD_CIVISIBILITY_AGENTLESS_ENABLED` (Required)

> : Enables or disables the agentless mode.<br/>

Reviewer: ```suggestion
: Enables or disables Agentless mode.<br/>
```

**Default**: `false`

`DD_API_KEY` (Required)
: The [Datadog API key][6] used to upload the test results.<br/>
**Default**: `(empty)`

> Then, prefix your test command with `dd-trace ci run`, providing the name of the service or library under test as the `--dd-service` parameter, and the environment where tests are being run (for example, `local` when running tests on a developer workstation, or `ci` when running them on a CI provider) as the `--dd-env` parameter. For example:

Reviewer: ```suggestion
Then, prefix your test command with `dd-trace ci run`. Providing the name of the service or library under test as the `--dd-service` parameter. Provide the environment where tests are being run (for example, `local` when running tests on a developer workstation, or `ci` when running them on a CI provider) as the `--dd-env` parameter. For example:
```

{{< code-block lang="bash" >}}
dd-trace ci run --dd-service=my-dotnet-app --dd-env=ci -- dotnet test
{{< /code-block >}}

Alternatively, you can provide the [Datadog API key][6] using the `--api-key` parameter, for example:

{{< code-block lang="bash" >}}
dd-trace ci run --api-key <API KEY> --dd-service=my-dotnet-app --dd-env=ci -- dotnet test
{{< /code-block >}}

> When the `--api-key` is set, the agentless mode is automatically enabled.

Reviewer: ```suggestion
When the `--api-key` is set, Agentless mode is automatically enabled.
```

> Additionally, configure the Datadog site to use the selected one ({{< region-param key="dd_site_name" >}}):

Reviewer: ```suggestion
Additionally, configure which [Datadog site][7] to which you want to send data. Your Datadog site is {{< region-param key="dd_site_name" >}}.
```

## Agentless (Beta)

To instrument your test suite without requiring an Agent, configure the following environment variables:

`DD_CIVISIBILITY_AGENTLESS_ENABLED` (Required)
: Enables or disables Agentless mode.<br/>
**Default**: `false`

`DD_API_KEY` (Required)
: The [Datadog API key][6] used to upload the test results.<br/>
**Default**: `(empty)`

Then, prefix your test command with `dd-trace ci run`. Use the `--dd-service` parameter to provide the name of the service or library. Use the `--dd-env` parameter to provide the environment where tests are being run (`local` when running tests on a developer workstation, `ci` when running them on a CI provider, etc.) For example:

{{< code-block lang="bash" >}}
dd-trace ci run --dd-service=my-dotnet-app --dd-env=ci -- dotnet test
{{< /code-block >}}

Alternatively, you can provide the [Datadog API key][6] using the `--api-key` parameter, for example:

{{< code-block lang="bash" >}}
dd-trace ci run --api-key <API KEY> --dd-service=my-dotnet-app --dd-env=ci -- dotnet test
{{< /code-block >}}

When the `--api-key` is set, Agentless mode is automatically enabled.

> Additionally, configure which [Datadog site][7] to which you want to send data. Your Datadog site is {{< region-param key="dd_site_name" >}}.

Reviewer: ```suggestion
Additionally, configure which [Datadog site][7] to which you want to send data. Your Datadog site is: {{< region-param key="dd_site" >}}.
```
